backend/src/main/java/com/htecgroup/skynest/service/impl/BucketServiceImpl.java

```diff
@@ -103,8 +104,10 @@ public void deactivateLambda(UUID bucketId, LambdaType lambda) {
 
   @Override
   public List<BucketResponse> listAllBuckets() {
+
     List<BucketEntity> entityList =
         (List<BucketEntity>) bucketRepository.findAllByOrderByNameAscCreatedOnDesc();
+    entityList.stream().filter(b -> doesCurrentUserHaveViewPermissionOnBucket(b));
```

Comment:
I think you could also do
```
this::doesCurrentUserHaveViewPermissionOnBucket
```
instead of
```
b -> doesCurrentUserHaveViewPermissionOnBucket(b)
```

not sure which one looks cleaner

Reply:
I guess the first one